### DISCUSSION TITLE
haprefix is empty?

### DISCUSSION THREAD

**Discussion by Trilis29** [john30/ebusd#575] [Q&A]
Hi,

As you see in mqtt explorer shows haprefix empty, is it supposed to be this way?
Nevertheless HA autodiscovery works...

![haprefix](https://user-images.githubusercontent.com/51697921/158036931-5163c0c1-96ac-4ff4-abf2-1388b9a8fbd8.png)

**Comment by john30** [comment]
what are the arguments ebusd is started with?

**Reply by Trilis29** [reply]
EBUSD_OPTS="--scanconfig --accesslevel=* -d enh:/dev/ttyUSB0 -l /var/log/ebusd.log --address=ff --latency=10 --mqttport=1883 --mqtthost=192.168.1.23 --mqttuser=xxxxx --mqttpass=xxxxx--configpath=/etc/ebusd/ --mqttint=/etc/ebusd/mqtt-hassio.cfg --mqttjson"

**Reply by john30** [reply]
ah, when using the integration only, the prefix is missing. you can circumvent by using `--mqtttopic=...` for now

**Reply by Trilis29** [reply]
so just adding `--mqtttopic=homeassistant` should show under homeassistant topic?

**Reply by john30** [reply]
corrected with 7356871ceae36054be376c032c40fc7dc2bed8ab

**Reply by john30** [reply]
yes, but I'd not use homeassistant as prefix as this might have negative impact on your HA. the default would be "ebusd"

**Reply by Trilis29** [reply]
ok, will try ebusd

**Reply by Trilis29** [reply]
global still left under EMPTY topic.
![image](https://user-images.githubusercontent.com/51697921/159124928-e13d06aa-634e-4ce1-a380-dde72ecec62f.png)

**Reply by john30** [reply]
thats probably the retained messages from the run before. at least running=false is a hint to that
